src/AutoUI/models/helpers.ts

```diff
@@ -204,15 +249,54 @@ export const generateSchemaFromRefScheme = (
 	};
 };
 
+export const getRefSchema = (schema: JSONSchema, refSchemePrefix: string) => {
+	const refScheme = parseDescriptionProperty(schema, 'x-ref-scheme') as
+		| string
+		| null;
+	return refScheme
+		? (get(schema, `${refSchemePrefix}${refScheme}`) as JSONSchema) ?? schema
+		: schema;
+};
+
+export const getPropertyScheme = (
+	schemaValue: JSONSchema | JSONSchemaDefinition,
+) => {
+	const json = isJSONSchema(schemaValue) ? parseDescription(schemaValue) : null;
+	return json && ('x-foreign-key-scheme' in json || 'x-ref-scheme' in json)
+		? json['x-foreign-key-scheme'] ?? json['x-ref-scheme']
+		: null;
```

Comment:
Would this work?
```suggestion
	return json?.['x-foreign-key-scheme'] ?? json?.['x-ref-scheme'] ?? null;
```

Reply:
nice, yep now that I've added al checks and types it works 👍